File: src/main/java/ru/citeck/ecos/uiserv/domain/EvaluatorEntity.java
package ru.citeck.ecos.uiserv.domain;

import lombok.Data;

import javax.persistence.*;
import java.util.Objects;

@Data
@Entity
@Table(name = "evaluators")
public class EvaluatorEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "evaluators_seq_gen")
    @SequenceGenerator(name = "evaluators_seq_gen")
    protected Long id;

    @Column(name = "evaluator_id")
    private String evaluatorId;

    @Column(name = "type")
    private String type;

    @Lob
    @Column(name = "config_json")
    private String configJson;

    @Column(name = "inverse")
    private boolean inverse;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        EvaluatorEntity that = (EvaluatorEntity) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/java/ru/citeck/ecos/uiserv/domain/evaluator/repo/EvaluatorEntity.kt
package ru.citeck.ecos.uiserv.domain.evaluator.repo

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "evaluators")
class EvaluatorEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "hibernate_sequence")
    @SequenceGenerator(name = "hibernate_sequence")
    val id: Long? = null

    @Column(name = "evaluator_id")
    var evaluatorId: String? = null

    @Column(name = "type")
    var type: String? = null

    @Lob
    @Column(name = "config_json")
    var configJson: String? = null

    @Column(name = "inverse")
    var inverse = false

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as EvaluatorEntity
        return id == that.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }
}
